libs/ui-lib/lib/ocm/components/clusterConfiguration/types.ts

```diff
@@ -0,0 +1,45 @@
+import { FeatureId } from '../../../common';
+import {
+  NewFeatureSupportLevelData,
+  NewFeatureSupportLevelMap,
+} from '../../../common/components/newFeatureSupportLevels';
+
+export type FeatureIdPlatform = Extract<
+  FeatureId,
+  'NUTANIX_INTEGRATION' | 'VSPHERE_INTEGRATION' | 'EXTERNAL_PLATFORM_OCI'
+>;
+
+export type DisabledInfo = {
+  isDisabled: boolean;
+  disabledReason?: string;
+};
+
+export type ExternalPlatformIntegrationStateMap = Record<FeatureIdPlatform, DisabledInfo>;
+
+export const createExternalPlatformIntegrationStateMap = (
+  newFeatureSupportLevelContext: NewFeatureSupportLevelData,
+  featureIds: FeatureIdPlatform[],
+  cpuArchitecture?: string,
+  featureSupportLevelData?: NewFeatureSupportLevelMap | null,
+): ExternalPlatformIntegrationStateMap => {
+  const integrationStateMap: ExternalPlatformIntegrationStateMap = {} as Record<
+    FeatureId,
+    DisabledInfo
+  >;
+
+  featureIds.forEach((featureId) => {
+    integrationStateMap[featureId] = {
+      isDisabled: newFeatureSupportLevelContext.isFeatureDisabled(
```

Comment:
Seems redundant to check for the reason and if it's disabled.
I think it's best to just check for the reason, and if you want to return the extra boolean prop, base it on the fact that there is a reason.